crates/sui/src/ptb/ptb.rs

```diff
@@ -560,12 +572,44 @@ impl PTB {
             )
             .await?;
 
-        println!("Transaction executed");
+        if let Some(effects) = transaction_response.effects.as_ref() {
+            if effects.status().is_err() {
+                return Err(anyhow!(
+                    "PTB execution {}. Transaction digest is: {}",
+                    Pretty(effects.status()),
+                    effects.transaction_digest()
+                ));
+            }
+        }
+
+        let summary = {
+            let effects = transaction_response.effects.clone();
+            Summary {
+                digest: transaction_response.digest,
+                status: effects
+                    .as_ref()
+                    .ok_or_else(|| anyhow!("Error"))?
```

Comment:
Nit: Probably want better error message here if they can fail. If theyshould never fail I'd replace with an `expect`.

Reply:
Oops, forgot about that.